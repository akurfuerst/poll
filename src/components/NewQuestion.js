import { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { connect } from 'react-redux';
import { handleSaveQuestion } from '../actions/shared';
import { Button, Divider, Grid, Form, Segment, Header } from 'semantic-ui-react';

const NewQuestion = props => {
    const navigate = useNavigate();
    const [optionOne, setOptionOne] = useState('');
    const [optionTwo, setOptionTwo] = useState('');

    const handleChangeOption1 = (e) => {
        setOptionOne(e.target.value);
    };
    const handleChangeOption2 = (e) => {
        setOptionTwo(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();

        props.dispatch(handleSaveQuestion(optionOne, optionTwo)).then(
            () => {
                setOptionOne('');
                setOptionTwo('');
                navigate('/');
            }
        );

    };

    return (
        <div>
            <Header textAlign="center" size="huge">Create your own poll</Header>
            <Header textAlign="center" size="large" data-testid="headline">Would you rather</Header>

            <Form onSubmit={handleSubmit}>
                <Segment placeholder>
                    <Grid columns={2} stackable textAlign="center">
                        <Divider vertical>Or</Divider>

                        <Grid.Row verticalAlign="middle">
                            <Grid.Column>
                                <Form.Group widths="equal">
                                    <Form.Input fluid label="First option" value={optionOne} onChange={handleChangeOption1} />
                                </Form.Group>

                            </Grid.Column>

                            <Grid.Column>
                                <Form.Group widths="equal">
                                    <Form.Input fluid label="Second option" value={optionTwo} onChange={handleChangeOption2} />
                                </Form.Group>
                            </Grid.Column>
                        </Grid.Row>
                    </Grid>
                </Segment>

                <div className="submit-button">
                    <Button type="submit" disabled={!(optionOne && optionTwo)} primary>Create Question</Button>
                </div>
            </Form>
        </div>
    );
};

export default connect()(NewQuestion);
