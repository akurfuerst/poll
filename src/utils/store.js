import { configureStore } from '@reduxjs/toolkit';
import middleware from '../middleware';
import reducers from '../reducers';


export const store = configureStore({
    reducer: reducers,
    middleware
});
