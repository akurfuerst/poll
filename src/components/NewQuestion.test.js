import { render, screen } from '@testing-library/react';
import NewQuestion from './NewQuestion';
import React from 'react';
import { Provider } from 'react-redux';
import { store } from '../utils/store';
import { BrowserRouter as Router } from 'react-router-dom';

test('Check if headline is in component', () => {
    const component = render(
        <Router>
            <Provider store={store}>
                <NewQuestion />
            </Provider>
        </Router>
    );

    const headline = screen.getByTestId('headline');

    expect(headline).toBeInTheDocument();
    expect(component).toMatchSnapshot();
});
