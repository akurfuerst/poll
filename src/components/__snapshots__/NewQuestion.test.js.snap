// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if headline is in component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="ui huge center aligned header"
        >
          Create your own poll
        </div>
        <div
          class="ui large center aligned header"
          data-testid="headline"
        >
          Would you rather
        </div>
        <form
          class="ui form"
        >
          <div
            class="ui placeholder segment"
          >
            <div
              class="ui stackable center aligned two column grid"
            >
              <div
                class="ui vertical divider"
              >
                Or
              </div>
              <div
                class="middle aligned row"
              >
                <div
                  class="column"
                >
                  <div
                    class="equal width fields"
                  >
                    <div
                      class="field"
                    >
                      <label>
                        First option
                      </label>
                      <div
                        class="ui fluid input"
                      >
                        <input
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="column"
                >
                  <div
                    class="equal width fields"
                  >
                    <div
                      class="field"
                    >
                      <label>
                        Second option
                      </label>
                      <div
                        class="ui fluid input"
                      >
                        <input
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="submit-button"
          >
            <button
              class="ui primary disabled button"
              disabled=""
              tabindex="-1"
              type="submit"
            >
              Create Question
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="ui huge center aligned header"
      >
        Create your own poll
      </div>
      <div
        class="ui large center aligned header"
        data-testid="headline"
      >
        Would you rather
      </div>
      <form
        class="ui form"
      >
        <div
          class="ui placeholder segment"
        >
          <div
            class="ui stackable center aligned two column grid"
          >
            <div
              class="ui vertical divider"
            >
              Or
            </div>
            <div
              class="middle aligned row"
            >
              <div
                class="column"
              >
                <div
                  class="equal width fields"
                >
                  <div
                    class="field"
                  >
                    <label>
                      First option
                    </label>
                    <div
                      class="ui fluid input"
                    >
                      <input
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="column"
              >
                <div
                  class="equal width fields"
                >
                  <div
                    class="field"
                  >
                    <label>
                      Second option
                    </label>
                    <div
                      class="ui fluid input"
                    >
                      <input
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="submit-button"
        >
          <button
            class="ui primary disabled button"
            disabled=""
            tabindex="-1"
            type="submit"
          >
            Create Question
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
